### DISCUSSION TITLE
"Run test" in release but "Debug" in debug mode ?

### DISCUSSION THREAD

**Discussion by ThomasCartier** [rust-lang/rust-analyzer#18532] [Q&A] [answered]
Hi,
As the title says.
![Screenshot_20241119_135634](https://github.com/user-attachments/assets/083c5de6-22fe-455d-9254-dcc9a7ef410f)

Right now, I have to set the release in runnable args but I would like to avoid to go back and forth in the settings.
![Screenshot_20241119_093415](https://github.com/user-attachments/assets/13d0e7d1-350d-482b-88ee-c136cd34cf77)

Clicking `Debug` when `--release` is defined won't stop at the breakpoints.

Thanks

**Comment by attilarepka** [comment]
Hello @ThomasCartier,
I do not think this is possible, due to the nature of the `--release` flag.

When you run `cargo test`, by default  it uses the profile `[unoptimized + debuginfo]`, which let's you stop at breakpoints.

If you want to stuck with `--release` argument you need to override your `Cargo.toml` with [custom profiles](https://doc.rust-lang.org/cargo/reference/profiles.html#custom-profiles):
```toml
[profile.release]
debug = true
```
This essentially will run your `--release` tests with debug information included, however breakpoints still won't work due to optimizations.

Hope this helps.

**Reply by ThomasCartier** [reply]
Ok, so no clean way to have a release build/run when clicking Run test and a debug build/run when clicking Debug.

That would be really nice.

**Reply by attilarepka** [reply, marked as answer]
Unfortunately what you want is not possible in any clean way AFAIK.

**Reply by ThomasCartier** [reply]
Thank you!
